### PR TITLE
feat(pvc): Support Snapshotting

### LINKED ISSUE #102: feature: Dragonfly Snapshots through PVC's

### PR DESCRIPTION
**Pull request by Pothulapati**
Fixes #102

### CODE REVIEW

File: api/v1alpha1/dragonfly_types.go [hunk 1]
```diff
@@ -71,6 +71,23 @@ type DragonflySpec struct {
 	// +optional
 	// +kubebuilder:validation:Optional
 	ServiceAccountName string `json:"serviceAccountName,omitempty"`
+
+	// (Optional) Dragonfly pod security context
```
Comment (ashotland) [inline api/v1alpha1/dragonfly_types.go]: // wrong comment?

**Review by ashotland — APPROVED** [review]
Note tiny comment

File: internal/resources/resources.go [hunk 2]
```diff
@@ -150,6 +157,33 @@ func GetDragonflyResources(ctx context.Context, df *resourcesv1.Dragonfly) ([]cl
...
+			})
+
+			statefulset.Spec.Template.Spec.Containers[0].VolumeMounts = append(statefulset.Spec.Template.Spec.Containers[0].VolumeMounts, corev1.VolumeMount{
+				Name:      "df",
+				MountPath: "/dragonfly/snapshots",
+			})
+		}
+
+		statefulset.Spec.Template.Spec.Containers[0].Args = append(statefulset.Spec.Template.Spec.Containers[0].Args, "--dir=/dragonfly/snapshots")
+		if df.Spec.Snapshot.Cron != "" {
```
Comment (ashotland) [inline internal/resources/resources.go:188]: What happens if snapshot is performed while there is no pvc attached?

Comment (Pothulapati) [reply internal/resources/resources.go:188]: It tries to write into `/dragonfly/snapshots` but gets an error:

```
I20230927 08:48:43.706144     1 dfly_main.cc:779] Found 750.00MiB available memory. Setting maxmemory to 600.00MiB
I20230927 08:48:43.747648    14 uring_proactor.cc:157] IORing with 1024 entries, allocated 102720 bytes, cq_entries is 2048
I20230927 08:48:43.747834     1 proactor_pool.cc:147] Running 1 io threads
I20230927 08:48:43.748344     1 dfly_main.cc:409] Listening on admin socket any:9999
E20230927 08:48:43.752486     1 snapshot_storage.cc:91] Data directory error: No such file or directory for dir /dragonfly/snapshots
I20230927 08:48:43.753701    14 listener_interface.cc:97] sock[5] AcceptServer - listening on port 9999
I20230927 08:48:43.753713    14 listener_interface.cc:97] sock[6] AcceptServer - listening on port 6379
I20230927 08:49:03.978279    14 server_family.cc:1610] Replicating NO:ONE
W20230927 08:50:00.000119    14 server_family.cc:590] Failed to perform snapshot Permission denied: Failed to create directories
```

Should we disallow this clause?

Comment (Pothulapati) [reply internal/resources/resources.go:188]: Let's error out if there is no pvc with snapshot schedule being given.

**Pull request merged**